docs/developer/designs/topology-awareness/README.md

```diff
@@ -0,0 +1,287 @@
+    // Preferred constraints influence scoring but don't prevent scheduling.
+    Type TopologyConstraintType `json:"type,omitempty"`
+
+    // TopologyLevel defines the minimal level in the topology hierarchy
+    // that this constraint applies to (e.g., "rack", "zone", "datacenter").
+    // Jobs will be scheduled to maintain locality at this level when possible.
+    TopologyLevel string `json:"topologyLevel"`
+
+    // Topology specifies the name of the topology CRD that defines the 
+    // physical layout to use for this constraint. This allows for supporting
+    // multiple different topology configurations in the same cluster.
+    Topology string `json:"topology,omitempty"`
+}
+
+type PodGroupSpec struct {
+    // ...existing fields...
+    
+    // TopologyConstraint defines the topology constraints for this PodGroup
+    TopologyConstraint TopologyConstraint `json:"topologyConstraints,omitempty"`
+}
+```
+
+#### Pod Grouper Implementation
+
+The pod-grouper component will be enhanced to:
+
+1. Look for topology-related annotations on the top owner of pods
+2. Extract and validate topology constraint information
+3. Set the corresponding fields in the PodGroup that it creates
+4. Support both Required and Preferred constraints
+
+For pods that are part of the same job but have different topology requirements, the pod-grouper will use the most restrictive constraints to ensure all pods can be scheduled.
+
+### Integration with KAI-Scheduler
+
+The topology-aware scheduling will integrate with KAI-Scheduler's existing components:
+
+1. **Plugin Architecture**: Implement as a plugin that can be enabled/disabled
+2. **Scoring Mechanism**: Add topology-based scoring to node ranking
+3. **API Extensions**: Provide APIs for specifying topology preferences in workloads
+
+### User Experience
+
+Users will be able to:
+
+1. Define topology constraints in their workload specifications
+2. Set preferences for topology-aware placement
+3. Configure the relative importance of topology vs. other scheduling factors
+4. View topology information for their running workloads
+
+## Implementation Plan
+
+The implementation will proceed in phases:
+
+1. Basic topology representation and ResourceGroup CRD integration
+2. Implementing TopologyInfo
+2. Implementation of Approach 1 (Topology Tree with Ranking)
+3. Performance evaluation and optimization
+4. User experience improvements and documentation
+5. Implementation of Approach 2 (Simulation-based Evaluation)
+6. Performance evaluation and optimization
+7. User experience improvements and documentation
+
+## Alternatives Considered
+
+- Using existing Kubernetes topology mechanisms like topology spread constraints, pod affinity
+
+These alternatives were considered but rejected due to breaking away from what end users already know and expect them to do and the need to create something more limiting to allow the algorithms to be more simple.
+
+## Future Work
```

Comment:
Since these are not concrete, I wonder why mentioning them at all?

Reply:
:+1: